### PR TITLE
feature: introduced the filterVars sanitizer

### PR DESCRIPTION
**Pull request by gitmiro**
Hey Aron

Sorry, a bit late but here it comes: the filter_vars sanitizer =) I made use of your existing code in the trait to pass the filter_vars parameters to the sanitizer constructor. Not sure if there's a better way as the syntax of defining the filters looks quite verbose.

Another thing is validation. Would it make sense to validate the `filter` argument or should I just let it fail with php's default filter_vars error?

Any feedback is appreciated.

cheers
miro

### CODE REVIEW

File: src/Sanitizers/FilterVars.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        $this->options = $opts['options'] ?? null;
+    }
+
+    /**
+     * @param $input
+     * @return string
+     */
+    public function sanitize($input)
+    {
+        return filter_var($input, $this->filter, $this->options);
```
Comment (ArondeParon) [inline src/Sanitizers/FilterVars.php:29]: I now see that `filter_var` it's second default argument is `FILTER_DEFAULT`. Will this default still work if no filter is passed?

File: src/Sanitizers/FilterVars.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+use ArondeParon\RequestSanitizer\Contracts\Sanitizer;
+
+class FilterVars implements Sanitizer
+{
+
+    private $filter;
+    private $options;
+
+    /**
```
[the remark this answers is not in the document]
Comment (gitmiro) [reply src/Sanitizers/FilterVars.php:13]: /**
     * FilterVars constructor.       <=== you mean this section, right?
     * @param  array  $opts
     */

File: src/Sanitizers/FilterVars.php [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+     * FilterVars constructor.
+     * @param  array  $opts
+     */
+    function __construct(array $opts)
```
[the remark this answers is not in the document]
Comment (gitmiro) [reply src/Sanitizers/FilterVars.php:17]: ouuu man. I don't know. haha. it even makes the custom params array from inside the request class more readable. consider it done.

File: src/Sanitizers/FilterVars.php (hunk 1 above)
Comment (gitmiro) [reply src/Sanitizers/FilterVars.php:29]: I've added the default into the constructor

File: src/Sanitizers/FilterVars.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (ArondeParon) [reply src/Sanitizers/FilterVars.php:14]: Yeah

[the remark this answers is not in the document]
Comment (gitmiro) [reply src/Sanitizers/FilterVars.php]: Okay, I removed the docblock completely

File: tests/Sanitizers/SanizerTest.php [hunk 4]
```diff
@@ -53,4 +54,14 @@ public function test_remove_non_numeric_sanitizer()
         $output = $sanitizer->sanitize('test1234-134AC~');
         $this->assertEquals('1234134', $output);
     }
-}
\ No newline at end of file
+
+    public function test_strip_tags_with_filter_vars()
+    {
+        $opts = [
```
[the remark this answers is not in the document]
Comment (gitmiro) [reply tests/Sanitizers/SanizerTest.php]: tests have been added

[the remark this answers is not in the document]
Comment (ArondeParon) [reply tests/Sanitizers/SanizerTest.php]: Great work, thanks!

**Comment by ArondeParon** [comment]
This feature has now been released in https://github.com/ArondeParon/laravel-request-sanitizer/releases/tag/2.2.

Thanks again!

**Pull request merged**